sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -265,8 +265,8 @@ object FileFormat {
    * fields of the [[PartitionedFile]], and do have entries in the file's metadata map.
    */
   val BASE_METADATA_EXTRACTORS: Map[String, PartitionedFile => Any] = Map(
-    FILE_PATH -> { pf: PartitionedFile => pf.toPath.toString },
-    FILE_NAME -> { pf: PartitionedFile => pf.toPath.getName },
+    FILE_PATH -> { pf: PartitionedFile => pf.filePath.urlEncoded },
+    FILE_NAME -> { pf: PartitionedFile => new Path(pf.filePath.toUri.getRawPath).getName },
```

Comment:
is `pf.filePath.toUri.getRawPath.split("/").last` better?

Reply:
Hmm, maybe. I think that's basically what the path object is doing, but `split` is probably lighter weight. If you think that's safer I'm for it.